### PR TITLE
Session Length + Edited Topics

### PR DESCRIPTION
**Pull request by Olliesama**
Creates a function that gets the current session length and has Moni say different things depending on it.
This edits a few topics but is mostly a basis for future use.

I've left the good afternoon topic alone as it seems to have future plans.

### CODE REVIEW

File: Monika After Story/game/script-farewells.rpy [hunk 1]
```diff
@@ -224,6 +256,15 @@ label bye_prompt_to_class:
     $ persistent._mas_greeting_type = store.mas_greetings.TYPE_SCHOOL
     return 'quit'
 
+label bye_prompt_to_class_late:
+    m 2ekc "Umm...you've been here with me for quite a while [player]."
+    m 2ekd "Are you sure you've had enough rest for it?"
+    m 2eka "Makre sure you take it easy, okay?"
```
Comment (Oharo) [inline Monika After Story/game/script-farewells.rpy]:
```suggestion
    m 2eka "Make sure you take it easy, okay?"
```

File: Monika After Story/game/script-farewells.rpy [hunk 2]
```diff
@@ -259,6 +297,18 @@ init 5 python:
         eventdb=evhand.farewell_database
     )
 
+init 5 python:
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-farewells.rpy]: remove this `init 5 python` block, there already is one for the this farewell.

File: Monika After Story/game/script-farewells.rpy [hunk 3]
```diff
@@ -342,10 +342,32 @@ init 5 python:
...
+        if session_time < datetime.timedelta(minutes=20):
+            m 1eub "Aww, going already?"
+            m 1efp "You havent even been here for 20 minutes!"
+            m 3hksdlb "I'm just kidding, [player]."
+            m 2eka "You're so sweet for seeing me even when you have so little time."
+            m 2hub "I just want you to know I really appreciate that!"
+            m 2eka "Study hard [player], I'm sure you'll do great!"
+            m 2hua "See you when you get back!"
+        elif session_time < datetime.timedelta(hours=1):
+            m 2eua "Alright, Thanks for spending some time with me [player]!"
```
Comment (jmwall24) [inline Monika After Story/game/script-farewells.rpy]:
```suggestion
            m 2eua "Alright, thanks for spending some time with me, [player]!"
```

File: Monika After Story/game/script-farewells.rpy [hunk 4]
```diff
@@ -342,10 +342,32 @@ init 5 python:
...
+            m 2eka "I honestly wish it could have been longer...but you're a busy [guy]."
+            m 2hua  "Nothing is more important than a healthy education."
```
Comment (jmwall24) [inline Monika After Story/game/script-farewells.rpy]:
```suggestion
            m 2hua "Nothing is more important than a healthy education."
```

File: Monika After Story/game/script-farewells.rpy [hunk 5]
```diff
@@ -342,10 +342,32 @@ init 5 python:
...
+            m 3eub "Teach me something when you get back!"
+            m "See you soon!"
+        elif session_time < datetime.timedelta(hours=6):
+            m 1hua "Study hard, [player]!"
+            m 1eua "Nothing is more attractive than a [guy] with good grades."
+            m 1hua "See you later!"
+        else:
+            m 2ekc "Umm...you've been here with me for quite a while [player]."
```
Comment (jmwall24) [inline Monika After Story/game/script-farewells.rpy]:
```suggestion
            m 2ekc "Umm...you've been here with me for quite a while, [player]."
```

File: Monika After Story/game/script-farewells.rpy [hunk 6]
```diff
@@ -377,10 +398,34 @@ init 5 python:
...
 label bye_prompt_to_work:
+    $ session_time = mas_getSessionLength()
     if mas_isMoniNormal(higher=True):
-        m 1hua "Work hard, [player]!"
-        m 1esa "I'll be here for you when you get home from work."
-        m 1hua "Bye-bye!"
+        if session_time < datetime.timedelta(minutes=20):
+            m 2eka "Aww, okay! Just checking in on me before heading out?"
+            m 3eka "You must be really short on time if you're leaving already."
+            m "It was really sweet of you to see me, even when so busy!"
```
Comment (jmwall24) [inline Monika After Story/game/script-farewells.rpy]:
```suggestion
            m "It was really sweet of you to see me, even when you're so busy!"
```

File: Monika After Story/game/script-farewells.rpy [hunk 7]
```diff
@@ -377,10 +398,34 @@ init 5 python:
...
+            m 3hub "Work hard [player]! Make me proud!"
```
Comment (jmwall24) [inline Monika After Story/game/script-farewells.rpy]:
```suggestion
            m 3hub "Work hard, [player]! Make me proud!"
```

File: Monika After Story/game/script-farewells.rpy [hunk 8]
```diff
@@ -377,10 +398,34 @@ init 5 python:
...
+        elif session_time < datetime.timedelta(hours=1):
+            m 1hksdlb "Oh! Alright! I was starting to get really comfortable, Ahaha."
```
Comment (jmwall24) [inline Monika After Story/game/script-farewells.rpy]:
```suggestion
            m 1hksdlb "Oh! Alright! I was starting to get really comfortable, ahaha."
```

File: Monika After Story/game/script-farewells.rpy [hunk 9]
```diff
@@ -377,10 +398,34 @@ init 5 python:
...
+            m 1rusdlb "I was expecting us to be a here a bit longer but you're a busy [guy]!"
+            m 1eka "It was great seeing you, even if it wasn't as long as I wanted..."
+            m 1kua "But then if it were up to me I'd have you all day!"
+            m 1hua "I'll be here waiting for you to get back home from work!"
+            m "Tell me all about it when you get back!"
+        elif session_time < datetime.timedelta(hours=6):
+            m 2eua "Heading to work then [player]?"
```
Comment (jmwall24) [inline Monika After Story/game/script-farewells.rpy]:
```suggestion
            m 2eua "Heading to work then, [player]?"
```

File: Monika After Story/game/script-farewells.rpy [hunk 10]
```diff
@@ -377,10 +398,34 @@ init 5 python:
...
+            m 2tku "Maybe you can think of me if it becomes stressful..."
+            m 2esa "Just do your best! I'll see you when you get back!"
+            m 2eka "I know you'll do great!"
+        else:
+            m 2ekc "Oh...You've been here quite a while now...and now you're going to work?"
+            m 2rksdlc "I was hoping you'd rest before doing anything too big."
+            m 2ekc "Try not to overexert yourself, okay?"
+            m 2ekd "If you need to take a breather, don't be afraid to do so!"
+            m 3eka "Just come home to me happy and healthy."
+            m 3eua "Stay safe [player]!"
```
Comment (jmwall24) [inline Monika After Story/game/script-farewells.rpy]:
```suggestion
            m 3eua "Stay safe, [player]!"
```

File: Monika After Story/game/script-moods.rpy [hunk 11]
```diff
@@ -216,16 +216,32 @@ init 5 python:
...
+            m 2ekd "You saying that so soon after arriving must mean it's pretty bad."
+            m 2ekc "I know you wanted to spend some time with me and even though we've hardly been together today..."
+            m 2eka "I think you should go and get some rest."
+            m 1eka "I'll still be here when you get better."
+            m 1hua "You'll always be in my thoughts, see you soon sweetie!"
+        if session_time > datetime.timedelta(hours=3):
+            m 2wuo "[player]!"
+            m 2wkd "You haven't been ill this entire time, have you?"
+            m 2ekc "I really hope not, I've had lots of fun with you today but if you've been feeling bad this entire time..."
+            m 2rkc "Well...Just promise to tell me earlier next time."
```
Comment (jmwall24) [inline Monika After Story/game/script-moods.rpy]:
```suggestion
            m 2rkc "Well... Just promise to tell me earlier next time."
```

File: Monika After Story/game/script-topics.rpy [hunk 12]
```diff
@@ -4257,33 +4257,53 @@ label monika_morning:
...
+            m "I don't get why some people wake up in the middle of the day."
+            m 1lsc "It just seems so unproductive."
+            m "You'd have less time to do things and you might miss out on a lot of things."
+            m 1ekc "It could also be a sign that you're not taking good care of yourself."
+            m "You're not being careless with your health, are you [player]?"
+            m 1tkc "I wouldn't want you to get sick easily, you know."
+            m "I'd be really sad if you spent less time with me because you had a fever or something."
+            m 1eka "As much as I'd love to take care of you, I'm still stuck here."
+            m 4eka "So start trying to be an early-riser like me from now on, okay?"
+            m 1hua "The more time you spend with me, the more happy I'll be~"
```
Comment (jmwall24) [inline Monika After Story/game/script-topics.rpy]:
```suggestion
            m 1hua "The more time you spend with me, the happier I'll be~"
```

File: Monika After Story/game/script-topics.rpy [hunk 13]
```diff
@@ -4257,33 +4257,53 @@ label monika_morning:
...
+        else:
+            m 1eub "Good morning to you too!"
+            m 1tku "Except that it's {i}not{/i} morning, [player]!"
+            m 2tub "Are you maybe losing track of time?"
+            m "You better be careful! You might lose your entire day otherwise."
+            m 2hub "And then once the day has died we'll be saying 'good mourning'!"
```
Comment (jmwall24) [inline Monika After Story/game/script-topics.rpy]:
```suggestion
            m 2hub "And then once the day has died we'll be saying 'good morning'!"
```

File: Monika After Story/game/script-farewells.rpy (hunk 4 above)
Comment (multimokia) [inline Monika After Story/game/script-farewells.rpy]:
```suggestion
            m 2hua  "Nothing is more important than a good education."
```

File: Monika After Story/game/script-farewells.rpy [hunk 14]
```diff
@@ -342,10 +342,32 @@ init 5 python:
...
+            m "See you soon!"
+        elif session_time < datetime.timedelta(hours=6):
+            m 1hua "Study hard, [player]!"
+            m 1eua "Nothing is more attractive than a [guy] with good grades."
+            m 1hua "See you later!"
+        else:
+            m 2ekc "Umm...you've been here with me for quite a while [player]."
+            m 2ekd "Are you sure you've had enough rest for it?"
+            m 2eka "Make sure you take it easy, okay?"
+            m "If you start feeling pretty bad, I'm sure {i}one day{/i} off won't hurt."
```
Comment (multimokia) [inline Monika After Story/game/script-farewells.rpy]:
```suggestion
            m "If you're not feeling too well, I'm sure {i}one day{/i} off won't hurt."
```

File: Monika After Story/game/script-farewells.rpy [hunk 15]
```diff
@@ -377,10 +398,34 @@ init 5 python:
...
-        m 1esa "I'll be here for you when you get home from work."
-        m 1hua "Bye-bye!"
+        if session_time < datetime.timedelta(minutes=20):
+            m 2eka "Aww, okay! Just checking in on me before heading out?"
+            m 3eka "You must be really short on time if you're leaving already."
+            m "It was really sweet of you to see me, even when so busy!"
+            m 3hub "Work hard [player]! Make me proud!"
+        elif session_time < datetime.timedelta(hours=1):
+            m 1hksdlb "Oh! Alright! I was starting to get really comfortable, Ahaha."
+            m 1rusdlb "I was expecting us to be a here a bit longer but you're a busy [guy]!"
```
Comment (multimokia) [inline Monika After Story/game/script-farewells.rpy]:
```suggestion
            m 1rusdlb "I was expecting us to be a here a bit longer, but you're a busy [guy]!"
```

File: Monika After Story/game/script-farewells.rpy [hunk 16]
```diff
@@ -377,10 +398,34 @@ init 5 python:
...
+            m 2eka "The day may be good or bad...but if it becomes too much think of something nice!"
+            m 4eka "Every day, no matter how badly it's going ends after all!"
+            m 2tku "Maybe you can think of me if it becomes stressful..."
+            m 2esa "Just do your best! I'll see you when you get back!"
+            m 2eka "I know you'll do great!"
+        else:
+            m 2ekc "Oh...You've been here quite a while now...and now you're going to work?"
+            m 2rksdlc "I was hoping you'd rest before doing anything too big."
+            m 2ekc "Try not to overexert yourself, okay?"
+            m 2ekd "If you need to take a breather, don't be afraid to do so!"
```
Comment (multimokia) [inline Monika After Story/game/script-farewells.rpy]:
```suggestion
            m 2ekd "Don't be afraid to take a breather if you need to!"
```

File: Monika After Story/game/script-moods.rpy [hunk 17]
```diff
@@ -216,16 +216,32 @@ init 5 python:
...
-        m 3hub "Get well soon, my love!"
-    
+        if session_time < datetime.timedelta(minutes=20):
+            m 1ekd "Oh no, [player]..."
+            m 2ekd "You saying that so soon after arriving must mean it's pretty bad."
+            m 2ekc "I know you wanted to spend some time with me and even though we've hardly been together today..."
+            m 2eka "I think you should go and get some rest."
+            m 1eka "I'll still be here when you get better."
+            m 1hua "You'll always be in my thoughts, see you soon sweetie!"
+        if session_time > datetime.timedelta(hours=3):
```
Comment (jmwall24) [inline Monika After Story/game/script-moods.rpy]:
```suggestion
        elif session_time > datetime.timedelta(hours=3):
```

Comment (jmwall24) [inline Monika After Story/game/script-moods.rpy]:
```suggestion
        elif session_time > datetime.timedelta(hours=3):
```

Comment (jmwall24) [inline Monika After Story/game/script-moods.rpy]:
```suggestion
        elif session_time > datetime.timedelta(hours=3):
```

**Comment by ThePotatoGuy** [comment]
will wait for multi to approve before merging

**Pull request merged**